src/Microsoft.Extensions.Http/DependencyInjection/HttpClientBuilderExtensions.cs

```diff
@@ -80,10 +80,78 @@ public static IHttpClientBuilder AddHttpMessageHandlerBuilderOptions(this IHttpC
             {
                 throw new ArgumentNullException(nameof(configureBuilder));
             }
-            
+
             builder.Services.Configure<HttpClientFactoryOptions>(builder.Name, options => options.HttpMessageHandlerBuilderActions.Add(configureBuilder));
 
             return builder;
         }
+
+        public static IHttpClientBuilder AddTypedClient<T>(this IHttpClientBuilder builder)
+            where T : class
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.Services.AddTransient<T>(s =>
```

Comment:
nit: Does it need to be all service locatory?

Reply:
We could get it down to one call instead of two, but it would require more codes. Factory functions always lead to service resolvers when they are non-trivial